File: java-core-plugins/src/main/java/org/unlaxer/tramli/plugins/docs/DocumentationPlugin.java
package org.unlaxer.tramli.plugins.docs;

import org.unlaxer.tramli.FlowDefinition;
import org.unlaxer.tramli.FlowState;
import org.unlaxer.tramli.Transition;

public final class DocumentationPlugin {
    public <S extends Enum<S> & FlowState> String toMarkdown(FlowDefinition<S> definition) {
        StringBuilder sb = new StringBuilder();
        sb.append("# Flow Catalog: ").append(definition.name()).append("\n\n");
        sb.append("## States\n\n");
        for (S state : definition.allStates()) {
            sb.append("- `").append(state.name()).append("`");
            if (state.isInitial()) sb.append(" (initial)");
            if (state.isTerminal()) sb.append(" (terminal)");
            sb.append('\n');
        }
        sb.append("\n## Transitions\n\n");
        for (Transition<S> t : definition.transitions()) {
            sb.append("- `").append(t.from()).append(" -> ").append(t.to()).append("` via `");
            if (t.processor() != null) sb.append(t.processor().name());
            else if (t.guard() != null) sb.append(t.guard().name());
            else if (t.branch() != null) sb.append(t.branch().name());
            else sb.append(t.type());
            sb.append("`\n");
        }
        return sb.toString();
    }
}

File: ts-plugins/src/docs/documentation-plugin.ts
import type { FlowDefinition } from '@unlaxer/tramli';

/**
 * Generates a markdown flow catalog from a flow definition.
 */
export class DocumentationPlugin {
  toMarkdown<S extends string>(definition: FlowDefinition<S>): string {
    const lines: string[] = [];
    lines.push(`# Flow Catalog: ${definition.name}`);
    lines.push('');
    lines.push('## States');
    lines.push('');
    for (const state of definition.allStates()) {
      const config = definition.stateConfig[state];
      let suffix = '';
      if (config?.initial) suffix += ' (initial)';
      if (config?.terminal) suffix += ' (terminal)';
      lines.push(`- \`${state}\`${suffix}`);
    }
    lines.push('');
    lines.push('## Transitions');
    lines.push('');
    for (const t of definition.transitions) {
      const via = t.processor?.name ?? t.guard?.name ?? t.branch?.name ?? t.type;
      lines.push(`- \`${t.from} -> ${t.to}\` via \`${via}\``);
    }
    return lines.join('\n');
  }
}
